packages/app/src/runner/events/telemetry.ts

```diff
@@ -36,4 +38,61 @@ export const addTelemetryListeners = (Cypress) => {
       // TODO: log error when client side debug logging is available
     }
   })
+
+  const recordSpan = (spanState: 'start' | 'end', command: Cypress.CommandQueue, extendRecordSpanFn: (span?: Span) => void) => {
+    try {
+      const runnable = Cypress.state('runnable')
+      const test = Cypress.state('test')
+
+      const runnableType = runnable.type === 'hook' ? runnable.hookName : runnable.type
+
+      let span: Span | undefined
+
+      const spanName = `${runnableType}: ${command.attributes.name}(${command.attributes.args.join(',')})`
+
+      if (spanState === 'start') {
+        span = telemetry.startSpan({
+          name: spanName,
+          opts: {
+            attributes: {
+              spec: runnable.invocationDetails.relativeFile,
+              test: `test:${test.fullTitle()}`,
+              'runnable-type': runnableType,
+            },
+          },
+          isVerbose: true,
+        })
+      } else {
+        span = telemetry.getSpan(spanName)
```

Comment:
It looks like the SpanStates are `start` and `end`. Shouldn't this else block be closing the span, not getting it?
Should these state be `start` and `end`?

Reply:
the span is closed in the callback which also allows for additional attributes to be added.